pkg/cli/package.go

```diff
@@ -0,0 +1,199 @@
+// Copyright 2023 Chainguard, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cli
+
+import (
+	"bytes"
+	"crypto/sha256"
+	"encoding/hex"
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+	"text/template"
+	"time"
+
+	"chainguard.dev/apko/pkg/apk/fs"
+	"chainguard.dev/apko/pkg/apk/tarball"
+	"github.com/chainguard-dev/clog"
+	"github.com/psanford/memfs"
+	"github.com/spf13/cobra"
+)
+
+func Package() *cobra.Command {
+	var name, version, epoch, description, license, url, commit, outdir string
+	var archs, deps, provides, replaces []string
+	var providerPriority, replacesPriority int
+	cmd := &cobra.Command{
+		Use:     "package",
+		Short:   "Emit a package for the given directory, tarball or file.",
+		Example: `melange package <dir>`,
+		Args:    cobra.ExactArgs(1),
+		RunE: func(cmd *cobra.Command, args []string) error {
```

Comment:
should we hoist the actual package logic somewhere public that can be called by consumers?

I don't have the full context for this change, but it seems like it'd be useful for folks (us) using melange as a library

Reply:
Yeah, definitely. Ideally this could call something like `emitPackage(controlSection, dir)` which is shared with the usual melange flow. That's basically where I pulled this code from, it's just not as straightforward in that code as it is here.

Reply:
I would love to make all of this a little more idiomatically go...